src/CreatureSearch.js

```diff
@@ -0,0 +1,56 @@
+import { Component } from 'react';
+import './CreatureSearch.css';
+
+export default class CreatureSearch extends Component {
+
+  state = { nameSearch: '', sortField: '' }
+
+  handleNameFilter = ({ target }) => {
+    this.setState({ nameSearch: target.value });
+  }
+
+  handleSortFilter = ({ target }) => {
+    this.setState({ sortField: target.value });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.props.onSearch(this.state);
+  }
+
+  componentDidCatch(prevProp, prevState) {
```

Comment:
this should be componentDidUpdate. 
I'm amazed it works but just an FYI it should be a 
componentDidUpdate